forte/data/ontology/core.py

```diff
@@ -635,5 +637,207 @@ def index_key(self) -> int:
         return self.tid
 
 
+class Grid:
+    """
+    Regular grid with a grid configuration dependent on the image size.
+    It is a data structure used to retrieve grid-related objects such as grid
```

Comment:
One of the main purposes of having a `Grid` is to use it to index shapes, let's highlight this.

On the other hand, what's the use case of retrieving a grid cell?

Reply:
https://github.com/hepengfe/forte/blob/24afd2f516ea104ca55a21e422e62a1e334895d3/forte/data/ontology/core.py#L787 I have rewrite the function that it can be used to compute whether there is overlapping area between the grid cell and the image.

Reply:
No, I mean compute the overlap of two irregular shape via grid.

Reply:
The full algorithm cannot be complete inside `Grid` alone.
The full algorithm is like below

stage1: build grid cell indices for each image   (use Grid)
```
for img_idx, img in enumerate(images):
      indices = grid.get_overlapped_grid_cell_indices(img) # returns grid cell indices the image is on
      for gc_idx in indices:
            d[gc_idx].append(image_idx)
```

stage2: search for candidate
```
indices = grid.build_index(img)
cand_pool = set()
for gc_idx in indices:
       cand_pool += set(d[gc_idx])
```

stage3: compute iou between the query image and images in cand_pool

Reply:
Looks like we need a tutorial or an example on this usecase for checking overlap between irregular shapes. Could you create a new issue on this?

Reply:
https://github.com/asyml/forte/issues/888

Reply:
We'll move the discussion to #888 and this problem shall be addressed in another PR.